<launch>
	<!-- ========================================================== -->
	<!-- Configuration of the Kinefly                               -->
	<!-- ========================================================== -->


    <param name="parameterfile"     type="string" value="$(env HOME)/kinefly1_phys.yaml" />
    <param name="filenameBackground" type="string" value="$(env HOME)/kinefly1_phys.png" />
    
    <param name="image_topic"        type="string" value="/camera1/image_raw" />
    <param name="n_queue_images"     type="int"    value="2" />
    <param name="n_edges_max"        type="int"    value="1" />
    <param name="rc_background"      type="double" value="100000" />
    <param name="scale_image"        type="double" value="1.0" />
    <param name="use_gui"            type="bool"   value="true" />
    <param name="wingbeat_min"       type="double" value="180" />
    <param name="wingbeat_max"       type="double" value="220" />
    
    <param name="head/autozero"      type="bool"   value="true" />
    <param name="head/threshold"     type="double" value="0.0" />
    <param name="head/tracker"       type="string" value="area" />   <!-- area, edge, or tip -->
    
    <param name="abdomen/autozero"   type="bool"   value="true" />
    <param name="abdomen/threshold"  type="double" value="0.5" />
    <param name="abdomen/tracker"    type="string" value="tip" />   <!-- area, edge, or tip -->
    
    <param name="left/autozero"      type="bool"   value="true" />
    <param name="left/threshold"     type="double" value="0.005" />
    <param name="left/tracker"       type="string" value="tip" />      <!-- area, edge, or tip -->
    <param name="left/saturation_correction"  type="bool"   value="false" />
    
    <param name="right/autozero"     type="bool"   value="true" />
    <param name="right/threshold"    type="double" value="0.005" />
    <param name="right/tracker"      type="string" value="tip" />      <!-- area, edge, or tip -->
    <param name="right/saturation_correction"  type="bool"   value="false" />
    
    <param name="aux/tracker"       type="string"  value="intensity" />
    
    
    <!-- LED Panels -->
    <param name="flystate2ledpanels/axis"                type="string" value="x" />
    <param name="flystate2ledpanels/method"              type="string" value="voltage" />
    <param name="flystate2ledpanels/mode"                type="string" value="velocity" />
    <param name="flystate2ledpanels/pattern_id"          type="int"    value="1" />

    <param name="flystate2ledpanels/coeff_voltage/adc0"  type="double" value="1.0" />
    <param name="flystate2ledpanels/coeff_voltage/adc1"  type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_voltage/adc2"  type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_voltage/adc3"  type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_voltage/funcx" type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_voltage/funcy" type="double" value="0.0" />

    <param name="flystate2ledpanels/coeff_usb/x0"        type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_usb/xha"       type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_usb/xhr"       type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_usb/xaa"       type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_usb/xar"       type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_usb/xl1"       type="double" value="1.0" />
    <param name="flystate2ledpanels/coeff_usb/xl2"       type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_usb/xr1"       type="double" value="-1.0" />
    <param name="flystate2ledpanels/coeff_usb/xr2"       type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_usb/xxi"       type="double" value="0.0" />
    
    <param name="flystate2ledpanels/coeff_usb/y0"        type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_usb/yha"       type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_usb/yhr"       type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_usb/yaa"       type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_usb/yar"       type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_usb/yl1"       type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_usb/yl2"       type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_usb/yr1"       type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_usb/yr2"       type="double" value="0.0" />
    <param name="flystate2ledpanels/coeff_usb/yxi"       type="double" value="0.0" />
    
    
    <!-- Voltage Channel 0 -->
    <param name="flystate2phidgetsanalog/v0enable"    type="bool"   value="true" />
    <param name="flystate2phidgetsanalog/v00"         type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v0ha"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v0hr"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v0aa"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v0ar"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v0l1"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v0l2"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v0lr"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v0r1"        type="double" value="1.0" />
    <param name="flystate2phidgetsanalog/v0r2"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v0rr"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v0xi"        type="double" value="0.0" />

    <!-- Voltage Channel 1 -->
    <param name="flystate2phidgetsanalog/v1enable"    type="bool"   value="true" />
    <param name="flystate2phidgetsanalog/v10"         type="double" value="-18.0" />
    <param name="flystate2phidgetsanalog/v1ha"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v1hr"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v1aa"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v1ar"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v1l1"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v1l2"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v1lr"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v1r1"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v1r2"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v1rr"        type="double" value="0.1" />
    <param name="flystate2phidgetsanalog/v1xi"        type="double" value="0.0" />

    <!-- Voltage Channel 2 -->
    <param name="flystate2phidgetsanalog/v2enable"    type="bool"   value="true" />
    <param name="flystate2phidgetsanalog/v20"         type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v2ha"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v2hr"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v2aa"        type="double" value="1.0" />
    <param name="flystate2phidgetsanalog/v2ar"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v2l1"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v2l2"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v2lr"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v2r1"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v2r2"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v2rr"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v2xi"        type="double" value="0.0" />

    <!-- Voltage Channel 3 -->
    <param name="flystate2phidgetsanalog/v3enable"    type="bool"   value="true" />
    <param name="flystate2phidgetsanalog/v30"         type="double" value="-18.0" />
    <param name="flystate2phidgetsanalog/v3ha"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v3hr"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v3aa"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v3ar"        type="double" value="0.1" />
    <param name="flystate2phidgetsanalog/v3l1"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v3l2"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v3lr"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v3r1"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v3r2"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v3rr"        type="double" value="0.0" />
    <param name="flystate2phidgetsanalog/v3xi"        type="double" value="0.0" />
    
    <param name="flystate2phidgetsanalog/autorange"   type="bool"   value="false" />
    <param name="flystate2phidgetsanalog/serial"      type="int"    value="277047" />


</launch>
